### ISSUE TITLE
Many to many inserting garbage values

### ISSUE THREAD

**Issue by emrosenf** [mikro-orm/mikro-orm#268]
All of my entities have assigned ids

```typescript
export abstract class BaseEntity implements IdEntity<BaseEntity> {
  @PrimaryKey({ type: 'text'})
  id = cuid();
  ...
}
```

I have a defined many-to-many relationship:

```typescript
export class User extends BaseEntity {
  @ManyToMany({ entity: () => RoleType, pivotTable: 'RoleType_to_User', joinColumn: 'B', inverseJoinColumn: 'A', })
  roles = new Collection<RoleType>(this)
  ...
}

export class RoleType extends BaseEntity {
  @ManyToMany({ entity: () => User, pivotTable: 'RoleType_to_User', joinColumn: 'A', inverseJoinColumn: 'B'})
  users = new Collection<User>(this)
  ...
}
```

When I add a `RoleType` to a `User`

```typescript
const role = await em.find(RoleType, { name: 'admin' })
user.roles.add(role)
```

Garbage values are inserted. See picture. (sqlite test environment).

<img width="248" alt="Screen Shot 2019-12-11 at 11 53 07 PM" src="https://user-images.githubusercontent.com/425523/70686264-ab9f3f80-1c60-11ea-973b-e2df8edb4c6e.png">

I believe this is related to [AbstractSqlDriver#80](https://github.com/mikro-orm/mikro-orm/blob/master/lib/drivers/AbstractSqlDriver.ts#L80)

```typescript
  async nativeInsert<T extends AnyEntity<T>>(entityName: string, data: EntityData<T>, ctx?: Transaction): Promise<QueryResult> {
    const collections = this.extractManyToMany(entityName, data);
    const pk = this.getPrimaryKeyField(entityName);
    const qb = this.createQueryBuilder(entityName, ctx, true);
    const res = await qb.insert(data).execute('run', false);
    res.row = res.row || {};
    // why not have data[pk] be the first option?
    res.insertId = res.insertId || res.row[pk] || data[pk];
    await this.processManyToMany(entityName, res.insertId, collections, ctx);

    return res;
  }
```

**Comment by B4nan** [comment]
Just to be sure - what values are generated with the cuid() function? What column do you consider being filled with garbage? 

Also btw the way you define the m:n now means both sides are owners. For modeling bidirectional m:n, you should link the other side at least from one entity. More info with examples here: https://mikro-orm.io/docs/relationships/#manytomany

**Comment by emrosenf** [comment]
`cuid` generates values that look like this: `ck42tg70c000r9xmidhm8da23`. Column `B` is not being filled with the `cuid` generated value, it is being filled with the insertion number. None of my `User` ids look like small integers.

I re-ran the test with the following relations and it still produced the same result:

```typescript

  @ManyToMany({ entity: () => RoleType, pivotTable: 'RoleType_to_User', joinColumn: 'B', inverseJoinColumn: 'A', owner: true,})
  roles = new Collection<RoleType>(this)

...

  @ManyToMany(() => User, user => user.roles)
  users = new Collection<User>(this)
```

**Comment by B4nan** [comment]
Got this one replicated. Looks like your proposed change works and no tests are broken.

**Comment by B4nan** [comment]
Fixed in `3.0.0-rc.4`
